lib/Data/Chronicle/Writer.pm

```diff
@@ -73,6 +74,12 @@ has [qw(cache_writer db_handle)] => (
     default => undef,
 );
 
+has 'ttl' => (
+    isa      => 'Int',
+    is       => 'ro',
+    required => 1
```

Comment:
required? I don't think it should be required. If it's set, then we set the TTL, and if not then there is no TTL. Then we gradually add TTLs throughout the code where Chronicle is used.